[HWKINVENT-168] no duplicate relationships + a couple of other fixes

metlos opened the pull request:
This contains a couple of things:
- fix for deleting data nodes of the data entities that were left behind during recursive delete
- fix for finding relationships by canonical path
- refactoring of the build so that the inventory unit tests run on both tinkergraph (quick) and Titan (distribution-compatible). This will make sure that we're testing with the same stuff with deploying with.
- finally, fix for https://issues.jboss.org/browse/HWKINVENT-168 itself - this was caused by the `linkWith()` method from generic relationships using custom code for creating them. This has now been modified to use the common code path with the rest of the codebase to ensure uniform behavior.

metlos committed:
Don't leave the data nodes behind when deleting recursively.

metlos committed:
Fix finding relationships by canonical path.

jkremser commented on hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/Util.java:
```diff
@@ -352,14 +356,15 @@ public static Query extendTo(TraversalContext<?, ?> context, Path path) {
+                });
+            };
+
+            categorizer.accept(entity);
```
is this line needed here?

metlos replied:
Yes, so that the `entity` ends up in the `deleted` or `verticesToDeleteThatDefineSomething` sets.

jkremser commented on hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop-titan-provider/src/test/resources/cassandra-config.yaml:
```diff
@@ -0,0 +1,106 @@
+
+commitlog_directory: ${project.build.directory}/cassandra/commitlog
+saved_caches_directory: ${project.build.directory}/cassandra/saved_caches
+data_file_directories: [${project.build.directory}/cassandra/data]
```
This file may diverge from http://git.io/vEZnq in the future, but I have no better solution.

Perhaps some Maven magic.

metlos replied:
IMHO that doesn't matter that much. All we care about is bringing Titan up with C*. It shouldn't matter too much for the itests if C* is configured differently.

metlos committed:
Rework the build so that the base inventory tests are run for each
tinkerpop backend, not just with the TinkerGraph.

This will make sure the behavioral differences between backends do not
slip through our testing.
